File: resources-server/src/main/java/work/fking/pangya/resources/HttpServerChannelInitializer.java
package work.fking.pangya.resources;

import io.netty.channel.Channel;
import io.netty.channel.ChannelInitializer;
import io.netty.channel.ChannelPipeline;
import io.netty.handler.codec.http.HttpObjectAggregator;
import io.netty.handler.codec.http.HttpServerCodec;
import io.netty.handler.stream.ChunkedWriteHandler;
import lombok.extern.log4j.Log4j2;

@Log4j2
public class HttpServerChannelInitializer extends ChannelInitializer<Channel> {

    private final HttpServerHandler httpServerHandler = new HttpServerHandler();

    @Override
    protected void initChannel(Channel channel) {
        ChannelPipeline pipeline = channel.pipeline();

        pipeline.addLast(new HttpServerCodec());
        pipeline.addLast(new HttpObjectAggregator(ResourcesServer.MAX_CONTENT_LENGTH));
        pipeline.addLast(new ChunkedWriteHandler());
        pipeline.addLast(httpServerHandler);
    }
}

File: resources-server/src/main/kotlin/work/fking/pangya/resources/HttpServerChannelInitializer.kt
package work.fking.pangya.resources

import io.netty.channel.Channel
import io.netty.channel.ChannelInitializer
import io.netty.handler.codec.http.HttpObjectAggregator
import io.netty.handler.codec.http.HttpServerCodec
import io.netty.handler.stream.ChunkedWriteHandler

class HttpServerChannelInitializer : ChannelInitializer<Channel>() {
    private val httpServerHandler = HttpServerHandler()

    override fun initChannel(channel: Channel) {
        val pipeline = channel.pipeline()
        pipeline.addLast(HttpServerCodec())
        pipeline.addLast(HttpObjectAggregator(ResourcesServer.MAX_CONTENT_LENGTH))
        pipeline.addLast(ChunkedWriteHandler())
        pipeline.addLast(httpServerHandler)
    }
}
